protocol/x/clob/keeper/untriggered_conditional_orders.go

```diff
@@ -286,11 +286,59 @@ func (k Keeper) MaybeTriggerConditionalOrders(ctx sdk.Context) (triggeredConditi
 				),
 			)
 		}
+
+		// Trigger conditional orders using the oracle price.
 		currentOraclePriceSubticksRat := k.GetOraclePriceSubticksRat(ctx, clobPair)
 		triggeredOrderIds := untriggeredConditionalOrders.PollTriggeredConditionalOrders(
 			currentOraclePriceSubticksRat,
 		)
 		triggeredConditionalOrderIds = append(triggeredConditionalOrderIds, triggeredOrderIds...)
+
+		// Trigger conditional orders using the last traded price.
+		perpetualId := clobPair.MustGetPerpetualId()
+		minTradePriceSubticks, maxTradePriceSubticks, found := k.GetTradePricesForPerpetual(ctx, perpetualId)
+		if found {
+			// Get the perpetual.
+			perpetual, err := k.perpetualsKeeper.GetPerpetual(ctx, perpetualId)
+			if err != nil {
+				panic(
+					fmt.Errorf(
+						"EndBlocker: untriggeredConditionalOrders failed to find perpetualId %+v",
+						perpetualId,
+					),
+				)
+			}
+
+			// Get the market param.
+			marketParam, exists := k.pricesKeeper.GetMarketParam(ctx, perpetual.Params.MarketId)
+			if !exists {
+				panic(
+					fmt.Errorf(
+						"EndBlocker: untriggeredConditionalOrders failed to find marketParam %+v",
+						perpetual.Params.MarketId,
+					),
+				)
+			}
+
+			// Calculate the max allowed range.
+			maxAllowedRange := lib.BigRatMulPpm(currentOraclePriceSubticksRat, marketParam.MinPriceChangePpm)
+			maxAllowedRange.Mul(maxAllowedRange, new(big.Rat).SetUint64(types.ConditionalOrderTriggerMultiplier))
+
+			upperBound := new(big.Rat).Add(currentOraclePriceSubticksRat, maxAllowedRange)
+			lowerBound := new(big.Rat).Sub(currentOraclePriceSubticksRat, maxAllowedRange)
+
+			for _, price := range []types.Subticks{minTradePriceSubticks, maxTradePriceSubticks} {
+				// Clamp the min and max trade prices to the upper and lower bounds.
+				clampedTradePrice := lib.BigRatClamp(
+					new(big.Rat).SetUint64(price.ToUint64()),
+					lowerBound,
+					upperBound,
+				)
+				triggeredOrderIds := untriggeredConditionalOrders.PollTriggeredConditionalOrders(clampedTradePrice)
```

Comment:
i feel pretty similarly, I think PollTriggeredConditionalOrders(oraclePrice, clampedMin, clampedMax) would be nice

Reply:
hmm I agree. I thought it was okay because we probably only have a few thousand conditional orders, so I didn't think it would make too much of a difference. but looking at the [CPU profile](https://app.datadoghq.com/profiling/search?query=env%3Atestnet%20service%3Avalidator&my_code=disabled&op_filter=focus_on%28function%3A%22EndBlocker%20%28abci.go%29%22%29&viz=flame_graph&start=1704923519491&end=1704927119491&paused=false), conditional order triggering does account for 60% of the CLOB Endblocker latency (still a small amount though). 

`PollTriggeredConditionalOrders(oraclePrice, clampedMin, clampedMax)` might not be the best solution since max/min prices do not necessarily exist (they are in the transient store). maybe sorting is the way to go?

let me address this in a separate PR since this PR is pretty huge.